### PR TITLE
Static comms again

### PR DESCRIPTION
**Pull request by stanalbatross**
## About The Pull Request

Adds back static comms to all maps,  two towers will spawn per game from a random list of towers, one near either LZ. Details not final yet.

## Why It's Good For The Game

Return of roleplay, making RTOs more useful, marines can no longer toss all their recently buffed cades onto the fob, gives harassers a nice place to attack.

## TO-DO

- [x] add back mini RTO packs
- [x] change around survie stuff

Clean up

- [x] LV
- [x] Big Red
- [x] Kutjevo
- [x] Annex
- [x] Shivas 
- [x] Trijent
- [ ] Sorokyne

Review locations for

- [ ] LV
- [ ] Big red
- [ ] kutjevo
- [ ] old prison
- [ ] annex
- [ ] old ice
- [ ] shivas
- [ ] trijent
- [ ] corsat
- [ ] soro

## Changelog

:cl: Stan_Albatross, Triiodine, Geeves
add: Static comms is back! Two towers will spawn in random locations every round, one near-ish to each LZ. They are also displayed on the tacmap. You must repair and power one of them to establish comms with the ship. Survivors can also do so, if they so wish. Sprites by Triiodine, landmark system by Geeves.
add: static comm towers will require specific comm frequencies to be added to them via multitool before they can be spoken on, this means survivors cannot use USCM comms before drop.
del: The Almayer no longer spawns with comms parts, and the comms parts crate cannot be bought.
fix: survivor radios now start turned off if they spawn in a body that will become a xeno.
qol: added in-editor camera and APC icons
qol: added broken, wires cut, and broken with no cell apc presets for mappers
fix: fixed a few poorly offset cams on the Almayer
/:cl:

### CODE REVIEW

**Comment by stanalbatross** [comment]
> Don't forget about Whiskey Outpost, it uses a prebuilt portable tcomms tower currently.

This doesn't touch the code for that

File: code/game/machinery/telecomms/presets.dm
Comment (Segrain) [inline code/game/machinery/telecomms/presets.dm:218]:
```suggestion
	if(HAS_TRAIT(I, TRAIT_TOOL_MULTITOOL))
		if(!toggled && (inoperable() || (health <= initial(health) * 0.5)))
			to_chat(user, SPAN_WARNING("\The [src.name] needs repairs to have frequencies added to its software!"))
			return
```
As is, it is impossible to actually _do those repairs_ if the tower is below half health, because entire `attackby()` is stopped before it can get to `..()` for welder's interaction.
...does it even need to check for `toggled` to ask for repairs, anyway? Seems like a copypaste from the existing `toggle_state()`, and at first glance it looks like it could be an `or` instead of an `and`:
```suggestion
	if(HAS_TRAIT(I, TRAIT_TOOL_MULTITOOL))
		if(inoperable() || (health <= initial(health) * 0.5))
			to_chat(user, SPAN_WARNING("\The [src.name] needs repairs to have frequencies added to its software!"))
			return
		if(!toggled)
			to_chat(user, SPAN_WARNING("\The [src.name] needs to be toggled on to have frequencies added to its software!"))
			return
```

**Review by Segrain — CHANGES_REQUESTED** [review]

File: code/game/machinery/telecomms/presets.dm
Comment (Segrain) [inline code/game/machinery/telecomms/presets.dm:204]: _Please_ have a `do_after()` of some length somewhere around here. It is being spammed for several screens' worth of logs at a time.

Comment (stanalbatross) [reply code/game/machinery/telecomms/presets.dm:210]: added

**Comment by c4xmaniac** [comment]
I only got to command once with static tcomms but I liked it, it makes planning more complex and gives you an objective to do and defend. The people that took it upon themselves to set up the defenses and hold them really enjoyed it. I think this is a good change and that more objectives are added to the maps to give the game more complexity.

**Comment by somepersondunno** [comment]
Apart from the tower itself sometimes being broken due to bugs with the power supply, this was a LOT of fun to play on. We had an awesome round yesterday on Big Red wherein we had to do a dual evac due to the tower being in general store while we were on LZ1, it was very HRP and tense.

**Comment by PurpleCIoud** [comment]
I love this static Comms. However a suggestion would be to make their locations be more map lore freindly. For instance on LV the SC is not even in a building. Even better. Is to have 1 SC in between 2 LZ but have more defencive structure around it.

I think That for the future if something like DEFCON is returned ( NOT TECWEBS), to have an option to buy a porta telecoms instead of req budget. That way command can chose to either secure tcoms for entire operation and lose budget, or take care of coms and have more equipment.

Static coms reduces the same gameplay every round and gives CiC more reliance on brain rather than being ARES with human mistakes

**Comment by stanalbatross** [comment]
> Apart from the tower itself sometimes being broken due to bugs with the power supply, this was a LOT of fun to play on. We had an awesome round yesterday on Big Red wherein we had to do a dual evac due to the tower being in general store while we were on LZ1, it was very HRP and tense.

can you tell me exactly on what maps it was broken

**Comment by IngloriouslyBad** [comment]
I like the move towards creating objectives, and I like the idea of giving engineers more to do than trying to do FOB work. I love that we are seeing people take interests in learning how to do basic engineer work again, where last week you'd be hard pressed to find someone outside of a veteran of CM who knew how to fix APCs and do power. This is good! However, we need more changes elsewhere to make it viable for someone to willingly or even forced OOCly to hold the objectives like TCOMS, Sensor towers, or power station for a planetside med bay to be operational. No one has wanted to lose out on their round getting those things objectives online in the past few years, so jumping to something that is pretty much mandatory during this test has shown what we already expected to happen and to those who've never been around for the previous power/tcomms, what they can expect from now on... In the past week we've seen the Queen + hive just flood over TCOMs and power stations and perma kill or cap anyone there, over and over. It's nothing new, we've seen it happen in years past. And if it's going to go back to it, so be it, that's life and it's not new to quite a few of us. Atleast it will force people to listen to orders and support eachother and the objectives more. 
We need something to help those people who are for the moment still willing to virtually sacrifice their round for the greater good by volunteering to stand around and guard TCOMs. Bravo Fobbit syndrome with an exciting new twist. Anyways, they'll get wiped out eventually. If you don't repair it as a human player, you have no comms, and if you do repair it and are reluctant to make someone the next batch of sacrificial lamb and stay beind to guard TCOMs, you're not going to have comms for long. The drone/lurker/runner/warrior meta TCOMs gang that has camped the location the last few days of tests will just destroy it and kill or even worse cap the next poor engineer and their escort sent there on orders to fix and hold the area. Hey, maybe everyone who goes to die there will get a medal... consolation prize! Sure though, we can point out a few times where it held when a robusto group held it down, but this is the test phase where people are interested in doing something new for a change. Those robustos aren't going to sign up to sit at the TCOMs at their DND or UNO table every round and make the whole experience look interesting to RP or valiant last stand worthy of legend, so we can't see their action as what we can expect as standard. 

A queen main made an interesting point when discussing all this and used this to tie in their thoughts; resisting out of nests was removed because it forced babysitting an objective or being forced to lose out on an important part of xeno gameplay , captures. Now we're adding back in TCOMs that needs babysitting or you lose an integral part of marine gameplay, comms. Sparked a thought back to revisiting requiring the hive to have guards for captures and resisting out of nests for escape attempts.

If this is really an attempt to slow down the murder ball gameplay, bring more RP as well as improve RP, and also to spread each sides forces again to relieve the current Murder ball meta, then great! Lets do it, but we need compromises.

**Comment by somepersondunno** [comment]
> > Apart from the tower itself sometimes being broken due to bugs with the power supply, this was a LOT of fun to play on. We had an awesome round yesterday on Big Red wherein we had to do a dual evac due to the tower being in general store while we were on LZ1, it was very HRP and tense.
> 
> can you tell me exactly on what maps it was broken

Personally I can attest to Trijent and LV having this issue, wherein we had to get a portacomms tower spawned in.

**Comment by FoxyStalin** [comment]
I think the biggest thing that needs to be worked on is positions these show up in, like, popping up in Admin on Solaris is practically halfway through the map. 
Though, in theory, that might wind up being nice, I kinda liked the actual 'Forward' Operating bases like back when we used to make a decent cadeline in Tfort or when Bar was actually held and manned.    

Also; I repeat my suggestion to make the SL Tracker be able to track the Comms Towers. With the random placement, It can be hard to find them if no one's tacmapping

**Comment by Paperbell** [comment]
I like this. Missions aren't nearly as routine as before. I think the change makes marines stay close to and cooperate more with their squads. It would be cool if static comms could be used as a cas marker. This way dropships could observe how the static comms are doing and rappel people there. Though maybe this would just save you 2 seconds and some back and forth communication since at least one jtac person would always be at static comms.

**Comment by stanalbatross** [comment]
will rebase then reopen

**Comment by stanalbatross** [comment]
reopening to deconflict first

File: code/controllers/subsystem/atoms.dm
Comment (fira) [inline code/controllers/subsystem/atoms.dm:111]: No, that's not an SSatoms thing 😠 
These are core internals, you can't slap stuff like that here

File: code/controllers/subsystem/ticker.dm
Comment (fira) [inline code/controllers/subsystem/ticker.dm:250]: Make your own subsystem (or reuse init of another), don't strap things in SSatoms

File: code/game/objects/items/devices/radio/radio.dm
Comment (fira) [inline code/game/objects/items/devices/radio/radio.dm:71]: Mega Spagget alert

Have it turn off in /datum/job/antag/xenos/proc/transform_to_xeno or something, or bust the radio key, i don't know, but radio init scanning/checking for unrelated stuff is kinda no-no...

**Review by fira — CHANGES_REQUESTED** [review]
pls resolve spagget

**Comment by stanalbatross** [comment]
@Geevies ^

File: code/controllers/subsystem/ticker.dm
Comment (fira) [reply code/controllers/subsystem/ticker.dm:250]: Sorry this is roundstart

Just setup a signal then, but really, don't strap random stuff in SSatoms

**Comment by Wintermote** [comment]
Static comms have been a mostly good change for marines - it makes things more interesting, but their positions on the map and the defensibility of those positions are a problem. Marines don't have the ability to reinforce and supply telecoms in the middle of the map - A single boiler can essentially nullify any defenses marines create, let alone the overextended nature of these positions making them easy pickings for any other caste either directly attacking or just harassing reinforcements. You can't even rely on sentries because burrowers can completely bypass any unmanned cade lines and eliminate sentries without risk.

I think the issues with defensibility of static comms needs to be adequately addressed before adding them as a regular feature.

File: code/game/objects/items/devices/radio/radio.dm
Comment (stanalbatross) [reply code/game/objects/items/devices/radio/radio.dm]: geev fix

File: code/controllers/subsystem/ticker.dm
Comment (stanalbatross) [reply code/controllers/subsystem/ticker.dm]: geev fix

**Comment by ohmidg** [comment]
> Static comms have been a mostly good change for marines - it makes things more interesting to play around,
> 
> but their positions on the map and the defensibility of those positions are a problem. Marines don't have the ability to reinforce and supply telecoms in the middle of the map - A single boiler can essentially nullify any defenses marines create, let alone the overextended nature of these positions making them easy pickings for any other caste either directly attacking or just harassing reinforcements. You can't even rely on sentries because burrowers can completely bypass any unmanned cade lines and eliminate sentries without risk.
> 
> I think the issues with defensibility of static comms needs to be adequately addressed before adding them as a regular feature.


Hmmm.. yes, if only we had some sort of transport vehicle that could be used to safely transport supplies and marines to the static comms locations. Perhaps something armoured? That would also therefore be encouraged through this change to not frontline since it's not intended to do so and provide the RP that a good portion of the playerbase is missing for everyone involved??

I do agree though that if TCOMMS is in a certain location without being immensely fortified, xenos could just commit 10-15 minutes of their time and half or less of their force if numbers are great enough to wiping whoever is at static TCOMMS very easily while the other half briefly hold off a metaball. Less than 10-15 minutes even. 

Having it be away from the FOB is good and encourages the aforementioned mystery armoured vehicle to actually do something useful when it comes to resupplying munitions, materials and troops (as intended) and gives marines a cool little outpost where a variety of people could hang out and probably provide a plethora of other things I'm not thinking of. A good enough xeno force though, could probably keep TCOMMS pretty locked down unless there was some sort of counter. I'm not sure what that counter would be, maybe some way to make the route to static TCOMMS similar to hydro road on LV by placing it behind the usual front where it's only really prodded by harassers and if you lose it, marines are usually on their last legs anyway.

It would give bravo something else to do: helping loading supplies onto the mystery armoured transport, maintaining another little minifob which has its own set of problems and probably gives them more stuff to shoot, more roleplay, perhaps an actual reason to patrol the road and supply route to static TCOMMS due to its importance, instead of just being ignored and having marines picked off one by one like at hydro road. 

Would also give harassers on xenos something else to prod at. Maybe it'll even be a bit more fun than catching Funnyname "Bald" Dinglefart who's only just learned how to juggle MOU's yet in the middle of nowhere and trying to pounce.

Seems like a fun change all around for everyone, not just something else for the marines only and adds actual RP potential, more in depth logistics which is only really handled by the people who sign up to do something like that anyway (bravo, apc crew and the poor saps at req), so John and friends wouldn't be crying about things being too complex, while giving xenos a new strat to cut off and confuse the marine force and has them dedicate some of their metaball troops elsewhere to something fun. Also lets bravo actually get to shoot things while also indulging their fobbit tendencies.

File: code/game/machinery/telecomms/presets.dm
Comment (fira) [inline code/game/machinery/telecomms/presets.dm:184]: Mind that `/datum/proc/Destroy` and `/obj/structure/proc/destroy` are two different procs -- and i think what you want is the former

**Review by fira — CHANGES_REQUESTED** [review]

**Comment by stanalbatross** [comment]
I will resolve conflicts after IOs are merged because IOs will create conflicts with every mapfile

**Comment by NuggetsOnToast** [comment]
Static comms is cool and fun, anyone who says otherwise is wrong and should cope harder. Only advice is to finer tune a few locations of where the towers are actually placed on certain maps as they are a little goofy in some areas.

**Comment by stanalbatross** [comment]
> Static comms is cool and fun, anyone who says otherwise is wrong and should cope harder. Only advice is to finer tune a few locations of where the towers are actually placed on certain maps as they are a little goofy in some areas.

Can you elaborate

**Comment by stanalbatross** [comment]
shut up stale bot will TM this soon

**Comment by JediKing112** [comment]
-99999999999 this is the worst thing i've ever seen and completely destroys the entire round

**Comment by BKryth** [comment]
image

Cannot make a replacement APC in this location without making a new wall

please fix

**Comment by Segrain** [comment]
> image
> 
> Cannot make a replacement APC in this location without making a new wall
> 
> please fix

Or _with_ a new wall, for that matter - APC construction requires a `turn/open/floor`.

**Comment by BKryth** [comment]
> > image
> > Cannot make a replacement APC in this location without making a new wall
> > please fix
> 
> Or _with_ a new wall, for that matter - APC construction requires a `turn/open/floor`.

image
No still didnt work, already built a Reinforced wall and it still didnt work, i could not place an APC

**Comment by brjoaoed** [comment]
Screenshot_7234

LV LZ2 comms area needs to be fixed, its currently on a spot with poor firing lines, 1 tile choke, and with indestructible jungle walls, defending the tower or placing defenses there isn't possible.

**Comment by Listaless** [comment]
Static communications has a problem of drawing away marine forces to make sure you can actually still communicate with the rest of the force, So let's say you have bravo FOB, you now need a second or part of another squad to hold coms located in what ever part of the map and likely people attempting to deal with whatever lurkers/runners exist in the backline. Matter of the fact is it's hard to get people to hold FOB let alone a second FOB(in this case T-coms), hell it's hard to get people to BUILD a fob.
Rant over.

Actual Suggestion: add the option for req to buy portable T-Coms with the price going up the larger the population, this should mitigate low pop's struggle to defend T-coms with low numbers while being a cost at higher pop's if you can't afford to expand your FOB to attempt to connect it to T-coms(which is consider the only real viable defense of them)

**Comment by stanalbatross** [comment]
> Screenshot_7234
> 
> LV LZ2 comms area needs to be fixed, its currently on a spot with poor firing lines, 1 tile choke, and with indestructible jungle walls, defending the tower or placing defenses there isn't possible.

fixed

File: code/game/machinery/telecomms/presets.dm
Comment (Segrain) [inline code/game/machinery/telecomms/presets.dm:212]:
```suggestion
	var/current_state = on
	if(!do_after(user, 20, INTERRUPT_NO_NEEDHAND|BEHAVIOR_IMMOBILE, BUSY_ICON_FRIENDLY, src))
		return
	if(current_state != on)
		to_chat(user, SPAN_NOTICE("\The [src] is already turned [on ? "on" : "off"]!"))
		return
```
So that if two users are trying to turn on about simultaneously, then after first user turns it on, second user's attempt at turning on does not result in turning off.

**Comment by Segrain** [comment]
image

Trijent, 72:177. `turf/open/asphalt` instead of `open/floor`.

**Comment by stanalbatross** [comment]
> image
> 
> Trijent, 72:177. `turf/open/asphalt` instead of `open/floor`.

fixed and fixed space tile on bigred

**Review by morrowwolf — APPROVED** [review]
Code LGTM, needs mapping and sprite approval

**Review by Agameofscones — APPROVED** [review]
maps n' sprites look good to me

**Pull request merged**
